lib/rules/no-jquery-methods.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * @fileOverview Disallow the use of specific jQuery methods.
+ * @author Marc Lynch
+ */
+'use strict';
+
+/**
+ * Determines if this expression matches the method blacklist invocation.
+ * @param {ASTNode} node The identifier node.
+ * @returns {Boolean} Returns true if the expression matches, otherwise false.
+ */
+
+const BLACKLIST = [
+  'add', 'addBack', 'after', 'ajaxComplete', 'ajaxError', 'ajaxSend', 'ajaxStart',
+  'ajaxStop', 'ajaxSuccess', 'andSelf', 'before', 'bind', 'change', 'clearQueue',
+  'clone', 'contents', 'contextmenu', 'dblclick', 'delay', 'delegate', 'dequeue', 'detach',
+  'end', 'error', 'fadeIn', 'fadeTo', 'fadeToggle', 'finish', 'focusin', 'focusout', 'hover',
+  'insertAfter', 'insertBefore', 'keydown', 'keypress', 'keyup', 'last', 'load', 'mousedown',
+  'mouseenter', 'mouseleave', 'mousemove', 'mouseout', 'mouseover', 'mouseup', 'next',
+  'nextAll', 'nextUntil', 'not', 'parentsUntil', 'prepend', 'prependTo', 'prev', 'revUntil',
+  'promise', 'queue', 'removeData', 'removeProp', 'replaceAll', 'replaceWith', 'resize', 'scroll',
+  'scrollLeft', 'select', 'serialize', 'show', 'size', 'slice', 'slideDown', 'slideToggle', 'stop',
+  'submit', 'toArray', 'toggle', 'toggleClass', 'unbind', 'undelegate', 'unload', 'unwrap', 'wrap',
+  'wrapAll', 'wrapInner'
+];
+
+module.exports = {
+  docs: {
+    category: 'Best Practices',
+    recommended: true
+  },
+  meta: {
+    blacklist: BLACKLIST
+  },
+  create(context) {
+    return {
+      MemberExpression(node) {
+        let propertyName = node.property.name;
+        let blackListName = BLACKLIST.includes(propertyName) ? propertyName : false;
```

Comment:
Looking at this it seems like we'll wind up blacklisting any method that is in the blacklist whether it is part of a jQuery object or not. We should add tests to ensure that isn't the case.

Note: I feel like doing this properly will be difficult due to variables, e.g.:

```js
const myJQueryObj = this.$();
myJQueryObj.add(foo);
```